### PR TITLE
[GEOS-10528] Empty groupList tag remains after group deletion

### PR DESCRIPTION
**Pull request by ahmedababnehgeo**
[GEOS-10528](https://osgeo-org.atlassian.net/browse/GEOS-10528)

We noticed that, in when using XML storage for roles, pressing `Remove Selected and remove role associations`, truly removes the group and role associations, but on inspecting the `roles.xml` file, we found an empty user groups tag is still there.

This can create confusion for the system administrators, since the empty tag need not be there anymore.

In this PR, we remove the `groupList` in the above-mentioned case.
  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [x] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by ahmedababnehgeo** [comment]
relevant JIRA: https://osgeo-org.atlassian.net/browse/GEOS-10528

**Comment by ahmedababnehgeo** [comment]
@taba90 PR ready for review

File: src/main/src/main/java/org/geoserver/security/impl/AbstractRoleStore.java [hunk 1]
```diff
@@ -210,6 +210,9 @@ public void disAssociateRoleFromGroup(GeoServerRole role, String groupname) thro
             roles.remove(role);
             setModified(true);
         }
+        if (helper.group_roleMap.get(groupname) != null && roles != null && roles.size() == 0) {
```
Comment (aaime) [inline src/main/src/main/java/org/geoserver/security/impl/AbstractRoleStore.java:213]: Surprised the PMD rule on the usage of isEmpty did not catch this one, but to be consistent with the resto of the code:

```suggestion
        if (helper.group_roleMap.get(groupname) != null && roles != null && roles.isEmpty()) {
```

**Review by aaime — CHANGES_REQUESTED** [review]
Looks good, minor nitpick for consistency, see below.

**Comment by ahmedababnehgeo** [comment]
@aaime comment resolved

**Pull request merged**